test/unit/app/models/date_validation_test.rb

```diff
@@ -0,0 +1,41 @@
+require "test_helper"
+
+class DateValidationTest < ActiveSupport::TestCase
+  class StubBaseRecord
+    include ActiveModel::Model
+    include ActiveModel::Validations::Callbacks
+    attr_accessor :some_date
+  end
+
+  class StubModel < StubBaseRecord
+    include DateValidation
+    validates :some_date, presence: true
+    date_attributes :some_date
+  end
+
+  test "should be valid when date attribute is a valid date" do
+    model = StubModel.new(some_date: { 1 => 2023, 2 => 9, 3 => 10, 4 => 0, 5 => 0 })
```

Comment:
What do keys `4` and `5` represent in this hash – hours and minutes?

I'm not sure the `DateValidation` concern will handle DateTime values correctly right now – it only expects 3 parameters to be passed in, and it calls `Date.new(date[1], date[2], date[3])`.

Does this concern need to be made compatible with DateTime fields too?

Reply:
Because we never pass the result of the date instantiation on to the model we don't actually need to worry about it, the original time values will still be used, and other validation rules will get the original data. As you mentioned on Slack, if we were to use text inputs for the time fields we would need to validate those too.